apps/crn-frontend/src/shared-research/state.ts

```diff
@@ -1,30 +1,42 @@
 import {
+  ListResearchOutputResponse,
+  ResearchOutputResponse,
+} from '@asap-hub/model';
+import {
+  atom,
   atomFamily,
-  selectorFamily,
-  useRecoilValue,
   DefaultValue,
+  selector,
+  selectorFamily,
   useRecoilState,
+  useRecoilValue,
+  useSetRecoilState,
 } from 'recoil';
-import {
-  ListResearchOutputResponse,
-  ResearchOutputResponse,
-} from '@asap-hub/model';
-
+import { authorizationState } from '../auth/state';
+import { useAlgolia } from '../hooks/algolia';
 import {
   getResearchOutput,
   getResearchOutputs,
   ResearchOutputListOptions,
 } from './api';
-import { authorizationState } from '../auth/state';
-import { useAlgolia } from '../hooks/algolia';
+
+type RefreshResearchOutputListOptions = ResearchOutputListOptions & {
+  refreshToken: number;
+};
 
 const researchOutputIndexState = atomFamily<
   { ids: ReadonlyArray<string>; total: number } | Error | undefined,
-  ResearchOutputListOptions
+  RefreshResearchOutputListOptions
 >({
   key: 'researchOutputIndex',
   default: undefined,
 });
+
+const refreshResearchOutputListing = atom<number>({
```

Comment:
Perhaps this should be called refreshResearchOutputIndex?

Reply:
Done